components/net/http_loader.rs

```diff
@@ -1803,15 +1816,15 @@ fn prompt_user_for_credentials(
 
 /// [HTTP network fetch](https://fetch.spec.whatwg.org/#http-network-fetch)
 async fn http_network_fetch(
-    request: &mut Request,
+    fetch_params: &mut FetchParams,
     credentials_flag: bool,
     done_chan: &mut DoneChannel,
     context: &FetchContext,
 ) -> Response {
     let mut response_end_timer = ResponseEndTimer(Some(context.timing.clone()));
 
-    // Step 1
-    // nothing to do here, since credentials_flag is already a boolean
+    // Step 1: Let request be fetchParams’s request.
+    let request = &mut fetch_params.request;
 
     // Step 2
```

Comment:
the rest of this implementation needs updating since the spec steps and the implementation don't align anymore, but that's not relevant to this PR